codexdht/private/eth/p2p/discoveryv5/encoding.nim

```diff
@@ -307,7 +307,7 @@ proc encodeHandshakePacket*(rng: var HmacDrbgContext, c: var Codec,
 
   authdataHead.add(c.localNode.id.toByteArrayBE())
 
-  let ephKeys = ? KeyPair.random(rng)
+  let ephKeys = ? KeyPair.random(PKScheme.Secp256k1, rng)
```

Comment:
Interesting, is the random generator tuned for different key types?

Reply:
Yeah, it picks them in order if it's not explicitly defined. I was compiling without the `secp` compiler define and it was failing for me all over, took sometime to figure out. But it's much better to be explicit, since we also explicitly check for that scheme in a few places.

Reply:
That's good to know. And +1 for being explicit. Compile failures due to missing defines are a pain.